packages/flutter/test/material/list_tile_test.dart

```diff
@@ -2333,6 +2250,151 @@ void main() {
+                    }
+
+                    if (states.contains(MaterialState.selected)) {
+                      return selectedColor;
+                    }
+
+                    return defaultColor;
+                  }),
+                  title: const TestText('title'),
+                  subtitle: const TestText('subtitle') ,
+                );
+              },
+            ),
+          ),
+        ),
+      );
+    }
+
+    // Test disabled state.
+    await tester.pumpWidget(buildFrame());
+    RenderParagraph title = _getTextRenderObject(tester, 'title');
+    expect(title.text.style!.color, disabledColor);
+
+    // Test enabled state.
+    enabled = true;
+    await tester.pumpWidget(buildFrame());
+    await tester.pumpAndSettle();
+    title = _getTextRenderObject(tester, 'title');
+    expect(title.text.style!.color, defaultColor);
+
+    // Test selected state.
+    selected = true;
+    await tester.pumpWidget(buildFrame());
+    await tester.pumpAndSettle();
+    title = _getTextRenderObject(tester, 'title');
+    expect(title.text.style!.color, selectedColor);
+  });
+
+  testWidgets('ListTile.iconColor respects MaterialStateColor', (WidgetTester tester) async {
+    bool enabled = false;
+    bool selected = false;
+    const Color defaultColor = Colors.blue;
+    const Color selectedColor = Colors.green;
+    const Color disabledColor = Colors.red;
+    final Key leadingKey = UniqueKey();
+
+    Widget buildFrame() {
+      return MaterialApp(
+        theme: ThemeData(useMaterial3: true),
+        home: Material(
+          child: Center(
+            child: Builder(
+              builder: (BuildContext context) {
+                return ListTile(
+                  enabled: enabled,
+                  selected: selected,
+                  iconColor: MaterialStateColor.resolveWith((Set<MaterialState> states) {
+                    if (states.contains(MaterialState.disabled)) {
+                      return disabledColor;
+                    }
+
+                    if (states.contains(MaterialState.selected)) {
+                      return selectedColor;
+                    }
+
+                    return defaultColor;
+                  }),
+                  leading: TestIcon(key: leadingKey),
+                );
+              },
+            ),
+          ),
+        ),
+      );
+    }
+
+    Color iconColor(Key key) => tester.state<TestIconState>(find.byKey(key)).iconTheme.color!;
+
+    // Test disabled state.
+    await tester.pumpWidget(buildFrame());
+    expect(iconColor(leadingKey), disabledColor);
+
+    // Test enabled state.
+    enabled = true;
+    await tester.pumpWidget(buildFrame());
+    await tester.pumpAndSettle();
+    expect(iconColor(leadingKey), defaultColor);
+
+    // Test selected state.
+    selected = true;
+    await tester.pumpWidget(buildFrame());
+    await tester.pumpAndSettle();
+    expect(iconColor(leadingKey), selectedColor);
+  });
+
   group('Material 2', () {
```

Comment:
For all of these M2 tests, can we hard wire `useMaterial3: false` so that if we switch the default to true these tests will still work correctly?

Reply:
Eventually we will want to make it the default and we don't want to have to go back to all these tests and update them. If we have missed any, I will take care of them in a sweep, but new stuff we should try to make sure that no matter what the default is the tests should continue to work.